packages/forklift-console-plugin/src/modules/Providers/utils/components/DetailsPage/DetailItem.tsx

```diff
@@ -132,55 +162,53 @@ export const DescriptionTitle: React.FC<{ title: string }> = ({ title }) => (
+ * The content field's element and if editable, next to that appears a press-able inline
  * link edit button with the pencil icon, for triggering the onEdit callback.
  *
  * @component
  * @param {ReactNode} content - The field's content element.
  * @param {Function} onEdit - Function to be called when the button is clicked.
  */
-export const EditableContentButton: React.FC<{
+
+export const ContentField: React.FC<{
   content: ReactNode;
   onEdit: () => void;
-}> = ({ content, onEdit }) => (
-  <DescriptionListDescription>
-    {content}
-    <Button variant="link" isInline onClick={onEdit}>
-      <Pencil className="forklift-page-details-edit-pencil" />
-    </Button>
-  </DescriptionListDescription>
-);
-
-/**
- * Component for displaying a non-editable content.
- *
- * @component
- * @param {ReactNode} content - The content of the description.
- */
-export const NonEditableContent: React.FC<{ content: ReactNode }> = ({ content }) => (
-  <DescriptionListDescription>{content}</DescriptionListDescription>
-);
+}> = ({ content, onEdit }) =>
+  onEdit ? (
+    <DescriptionListDescription>
+      {content}
+      <Button variant="link" isInline onClick={onEdit}>
+        <Pencil className="forklift-page-details-edit-pencil" />
+      </Button>
+    </DescriptionListDescription>
+  ) : (
+    <DescriptionListDescription>{content}</DescriptionListDescription>
+  );
 
 /**
  * Type for the props of the DetailsItem component.
  *
  * @typedef {Object} DetailsItemProps
  * @property {string} title - The title of the details item.
- * @property {ReactNode} content - The content of the details item.
  * @property {ReactNode} [helpContent] - The content to display in the help popover.
  * @property {ReactNode} [showHelpIconNextToTitle] - if true, adding a help icon next to the title for displaying the help popover.
  *    If false, show the default Patternfly dashed line under the title.
  * @property {string[]} [crumbs] - Breadcrumbs for the details item.
- * @property {Function} [onEdit] - Function to be called when the edit button is clicked.
+ * @property {ReactNode} content - first/only field content to be displayed for the details item.
+ * @property {ReactNode[]} contents - Array of content fields (starting from 2nd and forward) to be displayed for the details item.
+ * @property {Function[]} onEdit - function for the first/only content field, to be called when the edit button is clicked or null if the field is non editable.
+ * @property {Function[]} onEdits - Array of functions per content field (starting from 2nd and forward), to be called when the edit button is clicked or null if the field is non editable.
  */
 export type DetailsItemProps = {
   title: string;
-  content: ReactNode;
   helpContent?: ReactNode;
   showHelpIconNextToTitle?: boolean;
   moreInfoLabel?: string;
   moreInfoLink?: string;
   crumbs?: string[];
+  content: ReactNode;
+  contents?: ReactNode[];
```

Comment:
```suggestion
  content: ReactNode | ReactNode[];
```

and then when using it, ensure it'a an array before using it, for example by using a helper method:

``` ts
// Function to ensure the content is always an array of ReactNode
function ensureArray(content: ReactNode | ReactNode[]): ReactNode[] {
  if (Array.isArray(content)) {
    return content;
  } else {
    return [content]; // Wrap the single ReactNode in an array
  }
}
```

Reply:
I already saw that this solution used by you in our code for the `TableLabelCell` labels:  
https://github.com/kubev2v/forklift-console-plugin/blob/e00a3ddf5b4adbe06f4c267b17b9d2f70cddc29f/packages/forklift-console-plugin/src/modules/Providers/utils/components/TableCell/TableLabelCell.tsx#L55-L57

I less liked this since dividing to a content and an array of contents implies that there should be at least one content for the `DetailsItem` instead of one array that can be null (unlike labels for the table). But you can still set `content` to `null` then it's anyway not really enforced.
OK, I'll change it